libbess-python/bess.py

```diff
@@ -32,188 +39,241 @@ class APIError(Exception):
+            'QueueInc': module_msg.QueueIncArg,
+            'QueueOut': module_msg.QueueOutArg,
+            'Queue': module_msg.QueueArg,
+            'RandomUpdate': module_msg.RandomUpdateArg,
+            'Rewrite': module_msg.RewriteArg,
+            'RoundRobin': module_msg.RoundRobinArg,
+            'SetMetadata': module_msg.SetMetadataArg,
+            'Sink': bess_msg.EmptyArg,
+            'Source': module_msg.SourceArg,
+            'Split': module_msg.SplitArg,
+            'Timestamp': bess_msg.EmptyArg,
+            'Update': module_msg.UpdateArg,
+            'VLANPop': bess_msg.EmptyArg,
+            'VLANPush': module_msg.VLANPushArg,
+            'VLANSplit': bess_msg.EmptyArg,
+            'VXLANDecap': bess_msg.EmptyArg,
+            'VXLANEncap': module_msg.VXLANEncapArg,
+            'WildcardMatch': module_msg.WildcardMatchArg,
+        }
+        arg_msg = proto_conv.dict_to_protobuf(arg, message_map[mclass])
+        request.arg.Pack(arg_msg)
+        return self._request(self.stub.CreateModule, request)
 
     def destroy_module(self, name):
-        return self._request_bess('destroy_module', name)
+        request = bess_msg.DestroyModuleRequest()
+        request.name = name
+        return self._request(self.stub.DestroyModule, request)
 
     def get_module_info(self, name):
-        return self._request_bess('get_module_info', name)
+        request = bess_msg.GetModuleInfoRequest()
+        request.name = name
+        return self._request(self.stub.GetModuleInfo, request)
 
     def connect_modules(self, m1, m2, ogate=0, igate=0):
-        return self._request_bess('connect_modules',
-                {'m1': m1, 'm2': m2, 'ogate': ogate, 'igate': igate})
+        request = bess_msg.ConnectModulesRequest()
+        request.m1 = m1
+        request.m2 = m2
+        request.ogate = ogate
+        request.igate = igate
+        return self._request(self.stub.ConnectModules, request)
 
     def disconnect_modules(self, name, ogate = 0):
-        return self._request_bess('disconnect_modules',
-                {'name': name, 'ogate': ogate})
-
-    def run_module_command(self, name, cmd, arg):
-        return self._request_module(name, cmd, arg)
+        request = bess_msg.DisconnectModulesRequest()
+        request.name = name
+        request.ogate = ogate
+        return self._request(self.stub.DisconnectModules, request)
+
+    def run_module_command(self, name, cmd, arg_type, arg):
+        request = bess_msg.ModuleCommandRequest()
+        request.name = name
+        request.cmd = cmd
+
+        all_classes = inspect.getmembers(module_msg, lambda c: inspect.isclass(c))
+        arg_classes = dict(all_classes)
+        arg_classes['EmptyArg'] = bess_msg.EmptyArg
+
+        arg_msg = proto_conv.dict_to_protobuf(arg, arg_classes[arg_type])
+        request.arg.Pack(arg_msg)
+
+        response = self._request(self.stub.ModuleCommand, request)
+        if response.HasField('other'):
```

Comment:
Is this temporary, or will we have to update here whenever we introduce new response types?